### PR TITLE
Use FoldingCell library for UITableView and UI refinements

### PR DESCRIPTION
**Pull request by kumuwu**
This is mostly copying from Ramotion's example project and debugging...
Oh but it cool

### CODE REVIEW

File: bacon/Base.lproj/Main.storyboard [hunk 1]
```diff
@@ -444,10 +515,13 @@
...
                         <viewLayoutGuide key="safeArea" id="2yH-Go-pk1"/>
                     </view>
+                    <connections>
+                        <outlet property="tableView" destination="Wxi-zi-XoU" id="0cE-oP-dMw"/>
+                    </connections>
                 </viewController>
                 <placeholder placeholderIdentifier="IBFirstResponder" id="uGh-dA-pXs" userLabel="First Responder" sceneMemberID="firstResponder"/>
             </objects>
-            <point key="canvasLocation" x="801" y="-558"/>
+            <point key="canvasLocation" x="800.78125" y="-558.3984375"/>
```
Comment (fterh) [inline bacon/Base.lproj/Main.storyboard:876]: Why did integers become floating point numbers? O.o

If these represent pixels, shouldn't they be strictly integers?

File: bacon/controller/TransactionsViewController.swift [hunk 2]
```diff
@@ -10,20 +10,105 @@ import UIKit
 
 class TransactionsViewController: UIViewController {
 
+    enum Const {
```
Comment (fterh) [inline bacon/controller/TransactionsViewController.swift:13]: Should these magic numbers be refactored into the constants file?

File: bacon/controller/TransactionsViewController.swift [hunk 3]
```diff
@@ -10,20 +10,105 @@ import UIKit
...
+
     override func viewDidLoad() {
         super.viewDidLoad()
+        setup()
+    }
+    private func setup() {
+        cellHeights = Array(repeating: Const.closeCellHeight, count: Const.rowsCount)
+        tableView.estimatedRowHeight = Const.closeCellHeight
+        tableView.rowHeight = UITableView.automaticDimension
+        //tableView.backgroundColor = UIColor(patternImage: #imageLiteral(resourceName: "background"))
```
Comment (fterh) [inline bacon/controller/TransactionsViewController.swift:31]: There's a commented-out line of code here

File: bacon/controller/TransactionsViewController.swift [hunk 4]
```diff
@@ -10,20 +10,105 @@ import UIKit
...
+        if #available(iOS 10.0, *) {
```
Comment (fterh) [inline bacon/controller/TransactionsViewController.swift:32]: Maybe refactor iOS 10.0 into constants file too

File: bacon/controller/TransactionsViewController.swift [hunk 5]
```diff
@@ -10,20 +10,105 @@ import UIKit
...
+            tableView.refreshControl = UIRefreshControl()
+            tableView.refreshControl?.addTarget(self, action: #selector(refreshHandler), for: .valueChanged)
+        }
+    }
 
-        // Do any additional setup after loading the view.
+    @objc func refreshHandler() {
```
Comment (fterh) [inline bacon/controller/TransactionsViewController.swift:38]: Same thing with magic numbers

File: bacon/controller/TransactionsViewController.swift [hunk 6]
```diff
@@ -10,20 +10,105 @@ import UIKit
...
+}
+
+extension TransactionsViewController: UITableViewDataSource, UITableViewDelegate {
+    func tableView(_: UITableView, numberOfRowsInSection _: Int) -> Int {
+        return 10
+    }
+
+    func tableView(_: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        guard case let cell as TransactionCell = cell else {
+            return
```
Comment (fterh) [inline bacon/controller/TransactionsViewController.swift:56]: Are we expecting this to sometimes happen? Maybe you can log here

File: bacon/controller/TransactionsViewController.swift [hunk 7]
```diff
@@ -10,20 +10,105 @@ import UIKit
...
+                self?.tableView.refreshControl?.endRefreshing()
+            }
+            self?.tableView.reloadData()
+        }
     }
+}
+
+extension TransactionsViewController: UITableViewDataSource, UITableViewDelegate {
+    func tableView(_: UITableView, numberOfRowsInSection _: Int) -> Int {
+        return 10
```
Comment (fterh) [inline bacon/controller/TransactionsViewController.swift:51]: Magic number

File: bacon/controller/TransactionsViewController.swift [hunk 8]
```diff
@@ -10,20 +10,105 @@ import UIKit
...
+            cell.unfold(false, animated: false, completion: nil)
+        } else {
+            cell.unfold(true, animated: false, completion: nil)
+        }
+
+        cell.number = indexPath.row + 1
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let rawCell = tableView.dequeueReusableCell(withIdentifier: "transactionCell", for: indexPath)
```
Comment (fterh) [inline bacon/controller/TransactionsViewController.swift:70]: Magic string

File: bacon/controller/TransactionsViewController.swift [hunk 9]
```diff
@@ -10,20 +10,105 @@ import UIKit
...
+        guard let cell = rawCell as? FoldingCell else {
+            return rawCell
+        }
+        let durations: [TimeInterval] = [0.26, 0.2, 0.2]
```
Comment (fterh) [inline bacon/controller/TransactionsViewController.swift:74]: Magic numbers

**Review by GilgameshTC — APPROVED** [review]

**Pull request merged**